### DISCUSSION TITLE
Cannot connect Budibase to MS SQL Server

### DISCUSSION THREAD

**Discussion by Emokores256** [Budibase/budibase#13631] [Help]
I have been trying to connect Budibase to SQL Server but all in vain. I have enabled the SQL Server Browser and SQL Server (SQLEXPRESS), however, I have failed to enable the SQL Server Agent (SQLEXPRESS). I have restarted the service and database instance several times. I keep getting the error

> Error creating datasource: Unable to connect: Failed to connect to localhost\SQLEXPRESS in 15000ms.

I have tried enabling and disabling the Encrypt option too but I get the same result.

[![error connecting Budibase to SQL Server][1]][1] 

I am using SQL Authentication to log in. On my system, I login successfully. I have also enabled the TCP/IP for SQLEXPRESS. Unfortunately, nothing works for me. Are there any other suggestions or ways I can get these two to connect?

[![error configuring SQL Server Agent (SQLEXPRESS)][2]][2]

  [1]: https://i.sstatic.net/f39CnW6t.png
  [2]: https://i.sstatic.net/3KRAyvfl.png

**Comment by andz-bb** [comment]
hey @Emokores256 

I converted this one to a discussion as we don't have a widespread bug around connecting to MS SQL server.

Are you running your Budibase instance inside docker-compose? You may need to use something like `host.docker.internal\\SQLEXPRESS` for this so that it can access your machine's local IP rather than the docker VM's.

Could you also check your firewall settings to see if the connection from docker could be getting blocked by it? Thanks

**Reply by Emokores256** [reply]
Thank you @andz-bb for your response. I am running Budibase from the browser. I have never used a docker container before.

**Comment by madukan** [comment]
I use Budibase and MySQL all on docker (separate Docker networks, through their own compose files). Issue I find is that when I change the host network, Budibase lose the database connection. For instance when I am on home Wifi, my Mac IP is one and then when I'm on Mobile Data my IP is another. So I use hostfile to mange this, manually I add the IP to a hostname that's consistent. When I test this with MySQLWorkbench, all is fine, meaning my hostname is working for both Wifi and Mobile-Data. Changed hostname mapping seem to be working fine.

However, when using Budibase, it can't seem to connect to the Mobile-data based hostname even when the IP is correct. Quite strange..
